refactor(Style): Move Style enums to enum class

anthonybilinski opened the pull request:
Based on https://github.com/qTox/qTox/pull/6560.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/qtox/qtox/6563)
<!-- Reviewable:end -->

anthonybilinski committed:
refactor(Style): Move Style enums to enum class

robinlinden commented on src/widget/style.cpp:
```diff
@@ -228,7 +229,7 @@ QFont Style::getFont(Style::Font font)
     };
 
-    return fonts[font];
+    return fonts[static_cast<int>(font)];
```
Unrelated to this PR, but `fonts` should probably be a map, not an array? :x Would make this less spooky. Especially since there's nothing around the Font enum to indicate that it can't be reordered or anything. :P

anthonybilinski replied:
Yeah agreed, moved it to a std::map.

anthonybilinski committed:
refactor(Style): Move fonts storage from array to std::map

Strongly enforces ordering instead of being based off comments, and logically
represents the relation better.

anthonybilinski commented:
Whoops close/open was just trying to get it to re-generate the diff based on the new destination branch, but no luck.

robinlinden commented on src/widget/style.cpp:
```diff
@@ -219,17 +219,17 @@ QFont Style::getFont(Font font)
     };
 
-    return fonts[static_cast<int>(font)];
+    return fonts[font];
```
`.at(font)`? `operator[]` will create a new element and return that if the key doesn't exist in the map. :P `.at()` will try to throw and explode instead.

This is why there's no const overload for operator[]. :x If you switch to .at() the map should be able to be const.

anthonybilinski replied:
I like exploding over UB, done :D And moved to const.
